include/libtorrent/aux_/vector.hpp

```diff
@@ -143,6 +143,20 @@ namespace libtorrent { namespace aux {
 		return {vec.data() + static_cast<type>(begin), vec.data() + static_cast<type>(end)};
 	}
 
+	// TODO: find a better place for this function
```

Comment:
I think there may be a better name too. there's a boost function called ``numeric_cast`` which does runtime checking